django/test/runner.py

```diff
@@ -421,6 +428,10 @@ def __init__(self, pattern=None, top_level=None, verbosity=1,
         self.parallel = parallel
         self.tags = set(tags or [])
         self.exclude_tags = set(exclude_tags or [])
+        if test_name_patterns:
+            self.test_name_patterns = {convert_test_name_pattern(p) for p in test_name_patterns}
```

Comment:
`{convert_test_name_pattern(p) for p in test_name_patterns} or None` ?

Reply:
Won't that iterate on `None` when `test_name_patterns` is `None`?

Reply:
Could be fixed with `{convert_test_name_pattern(p) for p in test_name_patterns or ()} or None`

Reply:
I find the `if` version more readable. If the concern is a later change that would not set `self.test_name_patterns`, I can make the assignment to `None` before the `if`?
```python
self.test_name_patterns = None
if test_name_patterns:
    self.test_name_patterns = {convert_test_name_pattern(p) for p in test_name_patterns}
```